### PR TITLE
[Native][AmplifyAPIGraphQL] doc updates

### PR DESCRIPTION
**Pull request by lawmicha**
*Description of changes:*
API docs for GraphQL
- migrate structure to Storage doc's getting started.
- consolidate to Todo schema (Android was using Blog schema)
- update iOS code snippets to use latest Model-based API calls
- updated Android code snippets to use Todo class instead of Blog
- reordered the sections to be the rest of mutation/query/subscriptions before concepts and configure authorization modes

Future section ideas
- "Authorizing API calls with Amplify Auth" for owner based auth models.
- "Advanced schemas" enums, lists, connections

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/graphqlapi/fragments/android/getting-started/20_installLib.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Add the following dependencies to your **app** build.gradle file and click "Sync Now" when asked:
+
+```groovy
+dependencies {
+    implementation 'com.amplifyframework:core:0.10.0'
+    implementation 'com.amplifyframework:aws-api:0.10.0'
+}
+```
+
+Also at the top of the same file, add this piece of code to support the Java 8 features Amplify uses:
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/android/getting-started/20_installLib.md:11]: I'd rather not duplicate this. Let's just point them to the project setup guide.

File: docs/lib/graphqlapi/fragments/android/getting-started/30_initapi.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    Amplify.configure(getApplicationContext());
+    Log.i("APIQuickstart", "Amplify configured with api plugin");
+} catch (Exception exception) {
+    Log.e("APIQuickstart", exception.getMessage(), exception);
+}
+```
+
+Upon building and running this application you should see the following in your console window:
+
+```bash
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/android/getting-started/30_initapi.md:17]: This isn't bash. Use `console`.

File: docs/lib/graphqlapi/fragments/android/getting-started/30_initapi.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+To initialize the Amplify API category, we are required to use the `Amplify.addPlugin()` method followed by `Amplify.configure()`.
+
+Add the following code to the bottom of your MainActivity `onCreate` method (ideally this would go in your Application class but this works for getting started quickly):
+
+```java
+try {
+    Amplify.addPlugin(new AWSApiPlugin());
+    Amplify.configure(getApplicationContext());
+    Log.i("APIQuickstart", "Amplify configured with api plugin");
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/android/getting-started/30_initapi.md:9]: Let's capitalize API here

File: docs/lib/graphqlapi/fragments/ios/getting-started/30_initapi.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    } catch {
+        print("Failed to initialize Amplify with \(error)")
+    }
+
+    return true
+}
+```
+Upon building and running this application you should see the following in your console window:
+
+```bash
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/ios/getting-started/30_initapi.md:25]: `console`

File: docs/lib/graphqlapi/fragments/native_common/getting-started/common.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
+The Amplify API category provides an interface for retrieving and persisting your model data. The API category comes with default built-in support for AWS AppSync. The Amplify CLI allows you to define your API and provision a GraphQL service with CRUD operations and real-time functionality. The Amplify AWS API plugin leverages [AWS AppSync](https://aws.amazon.com/appsync/).
+
+## Goal
+To setup and configure your application with Amplify API to create a Todo persisted in the backend.
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/native_common/getting-started/common.md:4]: Not sure I'd know what "a Todo" is without reading more. Can we generalize this for the reader? Such as:

> To setup and configure your application with Amplify API to save items in the backend.

File: docs/lib/graphqlapi/fragments/native_common/getting-started/common.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+? Do you want a guided schema creation? 
+    `Yes`
+? What best describes your project: 
+    `Single object with fields (e.g., “Todo” with ID, name, description)`
+? Do you want to edit the schema now? 
+    `No`
+```
+
+The guided schema creation will create a schema with the following:
+```
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/native_common/getting-started/common.md:44]: `graphql`

File: docs/lib/graphqlapi/fragments/android/getting-started/30_initapi.md (hunk 2 above)
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/android/getting-started/30_initapi.md:17]: this section that shows the logging has been removed completely for the new copy from Storage docs that contains the Java and Kotlin code

File: docs/lib/graphqlapi/fragments/android/getting-started/20_installLib.md (hunk 1 above)
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/android/getting-started/20_installLib.md:11]: this section has been revamped to latest version of Storage

File: docs/lib/graphqlapi/fragments/native_common/getting-started/common.md (hunk 6 above)
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/native_common/getting-started/common.md]: nice, thanks

File: docs/lib/graphqlapi/fragments/android/getting-started/50_createtodo.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+```java
+private void createBlog() {
+    Blog blog = Blog.builder()
+        .name("My first blog")
+        .build();
+
+    Amplify.API.mutate(
```
[the remark this answers is not in the document]
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/android/getting-started/50_createtodo.md:7]: will update this in a follow up PR

File: docs/lib/graphqlapi/fragments/ios/getting-started/30_initapi.md (hunk 4 above)
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/ios/getting-started/30_initapi.md:25]: [self-review] hmm let me double check

Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/ios/getting-started/30_initapi.md]: ah, auto-save was off, updated now

**Comment by lawmicha** [comment]
merging this in! will add the same changes over in REST API docs (https://github.com/aws-amplify/docs/pull/1808)

**Pull request merged**